Race condition in readLoop at muxer shutdown

We need to protect against a recv channel being closed at shutdown when passing a message through it.

```
panic: send on closed channel

goroutine 930 [running]:
github.com/blinklabs-io/gouroboros/muxer.(*Muxer).readLoop(0x4002375e80)
        /gomod-cache/github.com/blinklabs-io/gouroboros@v0.145.0/muxer/muxer.go:396 +0x60
created by github.com/blinklabs-io/gouroboros/muxer.New in goroutine 758
        /gomod-cache/github.com/blinklabs-io/gouroboros@v0.145.0/muxer/muxer.go:101 +0x130
```

--- a/muxer/muxer.go
+++ b/muxer/muxer.go
@@ -393,6 +393,10 @@ func (m *Muxer) readLoop() {
 			)
 			return
 		}
-		recvChan <- msg
+		select {
+		case <-m.doneChan:
+			return
+		case recvChan <- msg:
+		}
 	}
 }

